### PR TITLE
parse less than token rather than left shift in context of type argum…

### LINKED ISSUE #23996: Bad parsing behavior for generic arrow type arguments
```ts
// No parse error.
type Foo = ReturnType< <T>(x: T) => T>;

// Parse error.
type Bar = ReturnType<<T>(x: T) => T>;
```

**Expected**: No parse error, `Foo` and `Bar` are equivalent.
**Actual**: Parse errors, `Bar` ends up as nonsense (`any`) because of a parse error.

### PR DESCRIPTION
**Pull request by Kingwl**
…ents

Fixes #23996

### CODE REVIEW

File: tests/baselines/reference/api/typescript.d.ts [hunk 1]
```diff
@@ -3071,6 +3071,7 @@ declare namespace ts {
         scanJsxIdentifier(): SyntaxKind;
         scanJsxAttributeValue(): SyntaxKind;
         reScanJsxToken(): JsxTokenSyntaxKind;
+        reScanLessThanToken(): SyntaxKind;
```
Comment (DanielRosenwasser) [inline tests/baselines/reference/api/typescript.d.ts]: Disgustingly, the corresponding method for a greater-than is called `reScanGreaterToken`. Not sure if we should be consistent here.

**Comment by Kingwl** [comment]
☝

File: src/compiler/parser.ts [hunk 2]
```diff
@@ -1093,6 +1093,10 @@ namespace ts {
             return currentToken = scanner.reScanTemplateToken();
         }
 
+        function reScanLesserToken(): SyntaxKind {
```
Comment (RyanCavanaugh) [inline src/compiler/parser.ts]: `reScanLessThanToken`

**Comment by Kingwl** [comment]
🔼

**Pull request merged**